### PR TITLE
COMMERCE-9433 Default display page template for a product

### PR DESCRIPTION
**Pull request by alessiorendina**
https://issues.liferay.com/browse/COMMERCE-9433

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/asset/display/page/portlet/CPDefinitionAssetDisplayPageFriendlyURLResolver.java [hunk 1]
```diff
@@ -138,6 +158,45 @@ public String getActualURL(
 				params, requestContext);
 		}
 
+		if ((cpDisplayLayout != null) &&
+			(cpDisplayLayout.getType() ==
+				CPDisplayLayoutConstants.TYPE_LAYOUT_PAGE_TEMPLATE_ENTRY)) {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/asset/display/page/portlet/CPDefinitionAssetDisplayPageFriendlyURLResolver.java:163]: fetch again with right type

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/frontend/taglib/servlet/taglib/LayoutCPDefinitionDisplayLayoutSelectionScreenNavigationCategory.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	public boolean isVisible(User user, CPDisplayLayout cpDisplayLayout) {
+		if (cpDisplayLayout == null) {
+			return true;
+		}
+
+		PermissionChecker permissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		try {
+			if (_commerceCatalogModelResourcePermission.contains(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/frontend/taglib/servlet/taglib/LayoutCPDefinitionDisplayLayoutSelectionScreenNavigationCategory.java:100]: use layout permission checker

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/frontend/taglib/servlet/taglib/LayoutPageTemplateEntryCPDefinitionDisplayLayoutSelectionScreenNavigationCategory.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	public boolean isVisible(User user, CPDisplayLayout cpDisplayLayout) {
+		if (cpDisplayLayout == null) {
+			return true;
+		}
+
+		PermissionChecker permissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		try {
+			if (_commerceCatalogModelResourcePermission.contains(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/frontend/taglib/servlet/taglib/LayoutPageTemplateEntryCPDefinitionDisplayLayoutSelectionScreenNavigationCategory.java:105]: use layout page template permission checker

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:close

**Comment by andrea-ale-sbarra** [comment]
forwarded here https://github.com/liferay-commerce/liferay-portal/pull/3378

**Pull request closed without merging**